graph/tree_diameter/checker.cpp

```diff
@@ -0,0 +1,99 @@
+#include <iostream>
+#include "testlib.h"
+
+struct Result {
+  long long X;
+  int Y;
+  std::vector<int> path;
+};
+
+int main(int argc, char *argv[]) {
+
+  registerTestlibCmd(argc, argv);
+
+  int N = inf.readInt();
+
+  using edge = std::pair<int, int>;
+  std::map<edge, long long> edges;
+  for (int i = 0; i < N - 1; i++) {
+    int a = inf.readInt();
+    int b = inf.readInt();
+    long long c = inf.readInt();
+    if (a > b) {
+      std::swap(a, b);
+    }
+    
+    edges[std::make_pair(a, b)] = c;
+  }
+  
+  
+  Result res_ans;
+  Result res_submitted;
+
+  res_ans.X = ans.readLong();
+  res_ans.Y = ans.readInt();
+
+  res_submitted.X = ouf.readLong();
+  res_submitted.Y = ouf.readInt();
```

Comment:
```suggestion
  res_submitted.Y = ouf.readInt(1, N);
```

このあとサイズ=res_submitted.Yのvectorを作るので、ユーザーがres_submitted.Yとして10^9とか入れると壊れる可能性があるので

Reply:
修正しました